akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala

```diff
@@ -370,6 +370,16 @@ import akka.util.unused
         }
     }
   }
+
+  /**
+   * Test if a Graph is an empty Source.
+   * */
+  def isEmptySource(graph: Graph[SourceShape[_], _]): Boolean = graph match {
+    case source: scaladsl.Source[_, _] if source eq scaladsl.Source.empty        => true
+    case source: javadsl.Source[_, _] if source.asScala eq scaladsl.Source.empty => true
```

Comment:
would it be better with `if source eq javadsl.Source.empty`?
I think that would avoid `asScala` conversion for cases when it's not empty.

Reply:
But the default implementation of `asScala` seems just return the `delegate`, but maybe that will change one day, will change.

Reply:
what I mean is that the conversion was needed for any other javadsl Source that would fall through to the `case _`

Reply:
Ohch! that's true for pf, thanks.